### ISSUE TITLE
Admin capability to delete existing user ids and associated data

### ISSUE THREAD

**Issue by smorgan19** [OAGi/Score#1315]
The ability to delete existing users is crucial for scalability and ensure a quick response to end users experiencing authentication issues with the tool.  The inability to delete existing user accounts causes admin's to have create multiple accounts for a users with something,  For instance, a simple mistake in the username causes an admin to have to create another account for a user.  Another example, if a non-sso account that belongs to a user has a slightly different username then the sso user, the accounts can't be linked.  This adds more unused accounts to the platform and as the user totals grow these disables account could affect performance and require more resource in the future.  The inability to delete accounts, also delays an admin's ability to solve authentication issues.  SSO account experiencing authentication issues, cannot be deleted and a new sso account created to fix the authentication issue.  An SSO account with WIP bie's data cannot be dissociated with the account, through transferring or copying the bies.  If the SSO account's authentication errors require a deletion, we have no way of retrieving that data through the UI.  If an admin had the option dissociate the data from that user to themselves or another user, delete the account, and create new one, this would solve a number of authentication errors.

**Comment by shaildshah** [comment]
The ability to delete an account should definitely provide options/some sort of panel to transfer ownership and address conflicts. Deleting provides more flexibility with how we manage our internal users.

**Comment by kbserm** [comment]
@smorgan19 and all, we didn't allow user deletion b/c not only we have to change ownership of entities, it could complicate the WHO info in the history log as well.

@hakjuoh does the history log points to ID (database key) or Username? Maybe part of the solution is to allow user name change.

**Comment by hakjuoh** [comment]
> @smorgan19 and all, we didn't allow user deletion b/c not only we have to change ownership of entities, it could complicate the WHO info in the history log as well.
> 
> @hakjuoh does the history log points to ID (database key) or Username? Maybe part of the solution is to allow user name change.

Yes, deleting user IDs would cause many conflicts in both WHO columns and logs. We should take account of all potential problems could cause in
1) Code List
2) CC
3) Business Context
4) BIE

For the SSO authentication issue, we may solve that through disconnecting SSO account info from the Score user account without erasing data.
